### PR TITLE
Fix Windows testsuite and main management comments

### PR DESCRIPTION
**Pull request by lamby**
This resolves issues:

 * #1360 - Get the testsuite working on Window
 * #1365 - Test opal on AppVeyor (Windows)
 * #1361 - Ensure we can run the main management commands on Windows.

### CODE REVIEW

File: opal/tests/test_core_commandline.py [hunk 1]
```diff
@@ -158,7 +158,8 @@ def test_test(self):
         with patch.object(commandline.test_runner, 'run_tests') as rt:
             commandline.test(mock_args)
             mock_args_call = rt.call_args[0][0]
-            self.assertTrue(mock_args_call.userland_here.endswith("opal"))
+            self.assertTrue(mock_args_call.userland_here.endswith("opal"),
+                            "Not running tests from root directory")
```
Comment (davidmiller) [inline opal/tests/test_core_commandline.py]: Nice.
I have never read the call signature of `assertTrue` closely enough to notice the second message.

**Comment by davidmiller** [comment]
This looks great, and demonstrably passes the tests :)

Can you talk me through the thinking behind https://github.com/openhealthcare/opal/commit/e918a6a0a6b110e0829127305a709790299b7ceb ? 

e.g. I can read what you're doing, but would like to understand why you're doing it...

File: opal/tests/test_core_commandline.py (hunk 1 above)
Comment (lamby) [reply opal/tests/test_core_commandline.py]: All the assertFoo calls take messages. Getting them phrased gramatically is a bit of PITA though as they "look" like comments in the code but aren't displayed in such a context when being run. :)

**Comment by lamby** [comment]
Re. e918a6a shall I comment there or here?

**Comment by davidmiller** [comment]
Wherever you like - I think I get notifications from everywhere ...

**Comment by lamby** [comment]
@davidmiller 

So the issue here was that Django doesn't *quite* support completely reloading ``django.conf.settings``. It somewhat pretends to, in that it can temporarily override settings and suchforth, but every attempt I have ever made at "please **move** to these settings over here" (ie. what we want to do here) doesn't work within the same Python interpreter. :)

The problems are mostly around things being imported twice (or other import side effects), but there are also problems about defaults - ie. if the current settings override a default setting but the "new" 


We basically bypass this by using a new global Python interpreter - we already are calling `manage.py` for a bunch of other things anyway so it would be no less ugly to do this and indeed, it makes might sense to do it like this anyway even if we were not having any settings issues atm. :)

File: opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2 [hunk 2]
```diff
@@ -13,9 +15,9 @@ class {{ name.title() }}Plugin(plugins.OpalPlugin):
     javascripts = {
         # Add your javascripts here!
         'opal.{{ name }}': [
-            # 'js/{{ name }}/app.js',
```
Comment (fredkingham) [inline opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2]: Is this right? I might be wrong but I think this should stay like it was

Comment (lamby) [reply opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2]: (on mobile) Might be a preprocessor/concatenator argument...

Comment (fredkingham) [reply opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2]: for context it get's passed to https://github.com/openhealthcare/opal/blob/v0.10.0/opal/templates/plugins/javascripts.html#L3

Comment (davidmiller) [reply opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2]: These strings end up passed to 
```
# ./opal/templates/plugins/javascripts.html
<script type="text/javascript" src="{% static variable %}"></script>
```
Does `{% static %}` take url paths or file paths ? I've never had to know... 
Looks like it's a URI to me?

Comment (lamby) [reply opal/scaffolding/plugin_scaffold/app/plugin.py.jinja2]: Fixed in https://github.com/openhealthcare/opal/commit/97d2b94672c3b959e2d2dd4941c66e99825dee19 :)

**Pull request merged**